python/remage/utils.py

```diff
@@ -1,7 +1,17 @@
 from __future__ import annotations
 
+from collections import defaultdict
+
 
 def _to_list(thing):
     if not isinstance(thing, tuple | list):
         return [thing]
     return thing
+
+
+def _icp_to_dict(thing):
```

Comment:
shouldn't we move this to ipc directly?

Reply:
what do you propose?

Reply:
I would add it on IpcResult (similar to get)